sympy/series/formal.py

```diff
@@ -0,0 +1,445 @@
+
+    No computation is performed.
+    This class should only to be used to represent
+    a series. No checks are performed.
+
+    For computing a series use :func:`fps`.
+
+    See Also
+    ========
+
+    sympy.series.formal.fps
+    sympy.series.formal.compute_fps
+    """
+    def __new__(cls, *args):
+        args = map(sympify, args)
+        return Expr.__new__(cls, *args)
+
+    @property
+    def function(self):
+        return self.args[0]
+
+    @property
+    def x(self):
+        return self.args[1]
+
+    @property
+    def x0(self):
+        return self.args[2]
+
+    @property
+    def ak(self):
+        return self.args[4][0]
+
+    @property
+    def xk(self):
+        return self.args[4][1]
+
+    @property
+    def ind(self):
+        return self.args[4][2]
+
+    @property
+    def interval(self):
+        return Interval(0, oo)
+
+    @property
+    def start(self):
+        return self.interval.inf
+
+    @property
+    def stop(self):
+        return self.interval.sup
+
+    @property
+    def length(self):
+        return oo
+
+    @property
+    def infinite(self):
+        """returns an infinite representation of the series"""
+        from sympy.concrete import Sum
+
+        ak, xk = self.ak, self.xk
+        k = ak.variables[0]
+
+        return self.ind + Sum(ak.formula * xk.formula, (k, ak.start, ak.stop))
+
+    def polynomial(self, n=6):
+        """truncated series as polynomial.
+
+        returns series sexpansion of f upto order ``O(x**n)``
+        as a polynomial (without ``O`` term)
+        """
+        terms = []
+        for i, t in enumerate(self):
+            if i >= n:
+                break
+            if t is not S.Zero:
+                terms.append(t)
+
+        return Add(*terms)
+
+    def truncate(self, n=6):
+        """truncated series.
+
+        returns truncated series expansion of f upto
+        order ``O(x**n)``
+
+        if n is ``None``, returns an infinite iterator
+        """
+        if n is None:
+            return iter(self)
+
+        x, x0 = self.x, self.x0
+        pt_xk = self.xk.coeff(n)
+        if x0 is S.NegativeInfinity:
+            x0 = S.Infinity
+
+        return self.polynomial(n) + Order(pt_xk, (x, x0))
+
+    def _eval_term(self, pt):
+        pt_xk = self.xk.coeff(pt)
+
+        def _get_xterm(t):
+            if self.x0 in [S.Infinity, -S.Infinity]:
+                t = t.as_numer_denom()[1]
+            else:
+                t = t.as_numer_denom()[0]
+            return t.as_independent(self.x)[1]
+
+        ind = S.Zero
+        if self.ind:
+            for t in Add.make_args(self.ind):
+                xterm = _get_xterm(t)
+                if pt_xk == 1 and xterm == 1:
+                    ind += t
+                elif xterm != 1 and pt_xk != 1:
+                    j = xterm.as_base_exp()[1]
+                    if j == pt:
+                        ind += t
+
+        try:
+            pt_ak = self.ak.coeff(pt).simplify() # TODO: Thoughts?
```

Comment:
I should have been more clear. TODO here was mainly due to the simplify. Just like you said I don't really like using it inside a function. But then again without simplify, the results tend to get very ugly.